libs/testserver/fake_workspace.go

```diff
@@ -190,6 +190,16 @@ func (s *FakeWorkspace) JobsCreate(request jobs.CreateJob) Response {
 	}
 }
 
+func (s *FakeWorkspace) JobsDelete(request jobs.DeleteJob) Response {
+	defer s.LockUnlock()()
+
+	delete(s.jobs, request.JobId)
```

Comment:
This could probably be replaced by MapDelete as well?

Reply:
`func MapDelete[T any](w *FakeWorkspace, collection map[string]T, key string) Response`

accepts `map[string]T`, while s.jobs is a `map[int64]jobs.Job`

Would you like to make `MapDelete` more generic?

Reply:
Thanks, I did not realize it's a different key type.

> Would you like to make MapDelete more generic?
Seems like a good idea, I don't see why not?

Reply:
>  Seems like a good idea, I don't see why not?
Done in the follow-up commit